File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.js
function solution(data, col, row_begin, row_end) {
    let answer = 0;
    
    data.sort((a, b) => {
        if(a[col - 1] === b[col - 1])
            return b[0] - a[0];
        
        return a[col - 1] - b[col - 1];
    });
    
    for(let i = row_begin; i <= row_end; i++) {
        let sum = 0;
        for(const d of data[i - 1]) {
            sum += d % i;
        }
        
        answer ^= sum;
    }
    
    return answer;
}

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int column;

bool Comp(vector<int> a, vector<int> b) {
    if(a[column] == b[column])
        return a[0] > b[0];
    
    return a[column] < b[column];
}

int solution(vector<vector<int>> data, int col, int row_begin, int row_end) {
    int answer = 0;
    column = col - 1;
    
    sort(data.begin(), data.end(), Comp);
    for(int i = row_begin; i <= row_end; i++) {
        int sum = 0;
        for(auto d: data[i-1]) {
            sum += d % i;
        }
        
        answer ^= sum;
    }
    
    return answer;
}